### ISSUE TITLE
static_transform_publisher with transient_local doesn't work

### ISSUE THREAD

**Issue by KenYN** [ros2/geometry2#487]
## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 20.04
- Installation type:
  - Galactic
- Version or commit hash:
  - Latest from apt-get
- DDS implementation:
  - CycloneDDS
- Client library (if applicable):
  - tf2_ros

#### Steps to reproduce issue
First, this is similar to #183, but the suggested resolution doesn't seem to work, as seen here; try the following two commands:

    $ ros2 run tf2_ros static_transform_publisher --x 1 --y 1 --z 1 --frame-id f1 --child-frame-id f2 &
    $ ros2 topic echo --qos-durability transient_local /tf_static

If I understand #183 correctly, the `--qos-durability transient_local` parameter will match up with the `static_transform_publisher`'s QoS and show me the transform.

#### Expected behavior
I expect to see output like this:

```
transforms:
- header:
    stamp:
      sec: 1639634982
      nanosec: 150475886
    frame_id: f1
  child_frame_id: f2
  transform:
    translation:
      x: 1.0
      y: 1.0
      z: 1.0
    rotation:
      x: 0.0
      y: 0.0
      z: 0.0
      w: 1.0
```

#### Actual behavior
All I see is the below and the program hangs waiting for messages:

`1639638202.723132 [0]       ros2: selected interface "lo" is not multicast-capable: disabling multicast`

#### Additional information
Note that I am using `loopback` due to multicast being blocked by my router, but I don't think that is connected.

I've also reported this [on ROS Answers](https://answers.ros.org/question/392875/static_transform_publisher-doesnt-latch-on-ros2/).

**Comment by clalancette** [comment]
You also need to add `--qos-reliability reliable` in Galactic.  On Rolling, that is no longer needed.  Does that work for you?

**Comment by KenYN** [comment]
Thanks, that fixed it!
